R/utils-tbl_regression.R

```diff
@@ -17,35 +17,62 @@
 #' or `broom.mixed::tidy`.
 #' @param conf.level confidence level passed directly to `broom::tidy`
 #' or `broom.mixed::tidy`.
+#' @param tidy_fun tidy function and arguments passed to it
 #' @noRd
 #' @keywords internal
 #' @author Daniel D. Sjoberg
 
 # Points function to use mixed vs non-mixed version of broom
-tidy_wrap <- function(x, exponentiate, conf.level) {
+tidy_wrap <- function(x, exponentiate, conf.level, tidy_fun) {
   mixed_classes <- c("lmerMod", "glmerMod", "nlme")
```

Comment:
It's been so long since the `broom.mixed` update I don't remember if there was something wrong with `nlme` back then or not. 

Not sure if this is new or not but the tidier for `nlme()` does have an additional column ("df") compared to the tidier for `lme4` objects - would that mess things up? 

``` r
fm1 <- nlme::nlme(
  height ~ SSasymp(age, Asym, R0, lrc),
  data = Loblolly,
  fixed = Asym + R0 + lrc ~ 1,
  random = Asym ~ 1,
  start = c(Asym = 103, R0 = -8.5, lrc = -3.3)
)
broom.mixed::tidy(fm1)
#> Registered S3 methods overwritten by 'broom.mixed':
#>   method         from 
#>   augment.lme    broom
#>   augment.merMod broom
#>   glance.lme     broom
#>   glance.merMod  broom
#>   glance.stanreg broom
#>   tidy.brmsfit   broom
#>   tidy.gamlss    broom
#>   tidy.lme       broom
#>   tidy.merMod    broom
#>   tidy.rjags     broom
#>   tidy.stanfit   broom
#>   tidy.stanreg   broom
#> Warning in tidy.lme(fm1): ran_pars not yet implemented for nonlinear models
#> # A tibble: 3 x 8
#>   effect group term  estimate std.error    df statistic  p.value
#>   <chr>  <chr> <chr>    <dbl>     <dbl> <dbl>     <dbl>    <dbl>
#> 1 fixed  fixed Asym    101.      2.46      68      41.2 7.89e-50
#> 2 fixed  fixed R0       -8.63    0.318     68     -27.1 3.35e-38
#> 3 fixed  fixed lrc      -3.23    0.0343    68     -94.4 7.81e-74


mod_glmer <-
  lme4::glmer(am ~ hp + (1 | gear), mtcars, family = binomial)
broom.mixed::tidy(mod_glmer)
#> # A tibble: 3 x 7
#>   effect   group term            estimate std.error statistic p.value
#>   <chr>    <chr> <chr>              <dbl>     <dbl>     <dbl>   <dbl>
#> 1 fixed    <NA>  (Intercept)       1.81      4.81       0.376   0.707
#> 2 fixed    <NA>  hp               -0.0121    0.0192    -0.633   0.527
#> 3 ran_pars gear  sd__(Intercept)   5.53     NA         NA      NA
```

<sup>Created on 2019-12-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
Oh it's actually a problem with `model.frame()` not outputting what we need for `nlme` in the utils `parse_fit()` function - I don't see a quick fix for now. 

``` r
fm1 <- nlme::nlme(
  height ~ SSasymp(age, Asym, R0, lrc),
  data = Loblolly,
  fixed = Asym + R0 + lrc ~ 1,
  random = Asym ~ 1,
  start = c(Asym = 103, R0 = -8.5, lrc = -3.3)
)



mod_glmer <-
  lme4::glmer(am ~ hp + (1 | gear), mtcars, family = binomial)

head(model.frame(fm1))
#> $reStruct
#> Random effects:
#>  Formula: Asym ~ 1 | Seed
#>             Asym Residual
#> StdDev: 5.078359        1
head(model.frame(mod_glmer))
#>                   am  hp gear
#> Mazda RX4          1 110    4
#> Mazda RX4 Wag      1 110    4
#> Datsun 710         1  93    4
#> Hornet 4 Drive     0 110    3
#> Hornet Sportabout  0 175    3
#> Valiant            0 105    3
```

<sup>Created on 2019-12-16 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
I'm removing `nlme` from the list of options in the `utils-tbl_regression` file for now.

Reply:
Added check to `parse_fit()` utility function to check if the `model.frame(x)` is a data frame with vector elements, and will output a message saying `tbl_regression` may be affected if it is not.